sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,130 @@
+                > note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            
+            > note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+        elt1, tok1 = element1
+        elt2, tok2 = element2
+        new_toks = 0
+        new_elt = Element()
+        new_elt.type = "Section"
+        # Merge binary representations by concatenation
+        if elt1.binary_representation == None or elt2.binary_representation == None:
+            new_elt.binary_representation = elt1.binary_representation or elt2.binary_representation
+        else:
+            new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
+        # Merge text representations by concatenation with a newline
+        if elt1.text_representation == None or elt2.text_representation == None:
+            new_elt.text_representation = elt1.text_representation or elt2.text_representation
+            new_toks = max(tok1, tok2)
+        else:
+            new_elt.text_representation = elt1.text_representation + '\n' + elt2.text_representation
+            new_toks = tok1 + 1 + tok2
+        # Merge bbox by taking the coords that make the largest box
+        if elt1.bbox == None and elt2.bbox == None:
+            pass
+        elif elt1.bbox == None or elt2.bbox == None:
+            new_elt.bbox = elt1.bbox or elt2.bbox
+        else:
+            new_elt.bbox = BoundingBox(
+                min(elt1.bbox.x1, elt2.bbox.x1),
+                min(elt1.bbox.y1, elt2.bbox.y1),
+                max(elt1.bbox.x2, elt2.bbox.x2),
+                max(elt1.bbox.y2, elt2.bbox.y2)
+            )
+        # Merge properties by taking the union of the keys
+        for k, v in elt1.properties.items():
+            new_elt.properties[k] = v
+        for k, v in elt2.properties.items():
+            if new_elt.properties.get(k) is None:
+                new_elt.properties[k] = v
+
+        return (new_elt, new_toks)
+        
+        
+
+    def merge_elements(self, document: Document) -> Document:
+        """Use self._should_merge and self._merge to greedily merge consecutive elements.
+        If the next element should be merged into the last 'accumulation' element, merge it.
+
+        Args:
+            document (Document): A document with elements to be merged. 
+
+        Returns:
+            Document: The same document, with its elements merged
+        """
+        if len(document.elements) < 2:
+            return document
+        token_counts = [len(self.tokenizer.encode(e.text_representation or "")) for e in document.elements]
```

Comment:
we have a tokenizer interface already, let's use that instead

Reply:
ack